### PR TITLE
MOSIP-35224 Moved config changes from dev integration release 13x

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: id-authentication-default.properties [hunk 1]
```diff
@@ -645,7 +650,7 @@ mosip.ida.vci.supported.cred.types=VerifiableCredential,MOSIPVerifiableCredentia
 
 # Regex to validate handles with provided key as the postfix
 # if the input handle is +855345353453@phone then the provided regex is used to validate the input.  
-mosip.ida.handle-types.regex={ '@phone' : '^\\+91[1-9][0-9]{7,9}@phone$' }
+mosip.ida.handle-types.regex={ '@phone' : '^\\+855[1-9][0-9]{7,9}@phone$' }
```
Comment (balaji-alluru) [inline id-authentication-default.properties]: This is camdgc specific property value, Pleaqse revert it back to previous value.

Comment (kameshsr) [reply id-authentication-default.properties]: Reverted.

**Review by balaji-alluru — APPROVED** [review]

File: ida-partner.cer [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+2Rt6FQ/naTQMRK/L19sQTvQgaYClfq9Yeuv1w449NFuJ/MixXdj3A/8OvGM6fgV7
+vHzZR8gSHNK9XpPfkyYUgXq+h2CHzQIDAQABoz8wPTAMBgNVHRMBAf8EAjAAMB0G
+A1UdDgQWBBSpx0wEBrp2hw1U8OMWzalKwaMn5DAOBgNVHQ8BAf8EBAMCBSAwDQYJ
+KoZIhvcNAQELBQADggEBAFzssGHBrBYV6OKY7PbFVge2B+SxvIRrUxFSnJIqxyGs
+mwgvEobzprH2pDir7xmQYNzG/npCw9980sJHxXRaojsby4+zxpECvTDa/00CVa3B
+4BQCumaF6aqXaY9/Eabijay9lmkKOc7robNyn0W0H39XciO0ECas+byK0EjXyvzw
+tUS2WZPnl1DXZBb6Z3yPZ/1e00dhga/IyDvBATR+++m69P0HHAOWOUKqtUDEfHTj
+kopRmdOKv70fgoXOGi0ucnCIqpJF6CiinR7PTsNXZ5IVTDQ3jdpmh9lL4Mi4gmPu
+F0SRhkXVig2LLe/OiQx7+0mjtPdKz4oEy1qeah+gCnY=
+-----END CERTIFICATE-----
```
Comment (vishwa-vyom) [inline ida-partner.cer]: what is this certificate, why is this required in the config server ?

File: mosip-vid-policy.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
             "vidType": "OneTimeUse",
             "vidPolicy": {
                 "validForInMinutes": null,
-                "transactionsAllowed": 1,
+                "transactionsAllowed": 2,
```
Comment (vishwa-vyom) [inline mosip-vid-policy.json:26]: why are we doing this change ?

File: partner-management-default.properties [hunk 4]
```diff
@@ -67,7 +67,7 @@ hibernate.generate_statistics=false
 spring.datasource.initialization-mode=never
 
 ## This property is used by kernel-authcodeflowproxy-api to check request is coming from allowed urls not.
-auth.allowed.urls=https://${mosip.pmp.host}/
+auth.allowed.urls=https://${mosip.pmp.host}/,https://${mosip.pmp.reactjs.ui.host}/
```
Comment (vishwa-vyom) [inline partner-management-default.properties]: I believe this change is related to the latest revamp, is that in the scope for the beta.1 release ?

Comment (vishwa-vyom) [reply partner-management-default.properties]: Lets understand this for all the changes this file

File: resident-default.properties [hunk 5]
```diff
@@ -348,7 +348,7 @@ mosip.iam.adapter.clientsecret=${mosip.resident.client.secret}
...
+mosip.service.end-points=/**/req/otp,/**/proxy/**,/**/validate-otp,/**/channel/verification-status/**,/**/req/credential/**,/**/req/card/*,/**/req/auth-history,/**/rid/check-status,/**/req/auth-lock,/**/req/auth-unlock,/**/req/update-uin,/**/req/print-uin,/**/req/euin,/**/credential/types,/**/req/policy/**,/**/aid/status,/**/individualId/otp,/**/mock/**,/**/callback/**,/**/download-card,/**/download/registration-centers-list/**,/**/download/supporting-documents/**,/**/vid/policy,/**/vid,/vid/**,/**/download/nearestRegistrationcenters/**,/**/authorize/admin/validateToken,/**/logout/user,/**/aid-stage/**
```
Comment (vishwa-vyom) [inline resident-default.properties:351]: Why do we need this change ?

File: resident-default.properties [hunk 6]
```diff
@@ -727,6 +729,7 @@ resident.template.summary.success.REVOKE_VID=gen-or-revoke-vid-success-summary
 resident.template.summary.success.GET_MY_ID=get-my-uin-card-success-summary
 resident.template.summary.success.VALIDATE_OTP=verify-my-phone-email-success-summary
 resident.template.summary.success.VID_CARD_DOWNLOAD=vid-card-download-positive-summary
+resident.template.summary.success.AUTHENTICATION_REQUEST=mosip.event.type.AUTHENTICATION_REQUEST
```
Comment (vishwa-vyom) [inline resident-default.properties]: why do you have the same value for summary, failure and success ?

File: resident-default.properties [hunk 7]
```diff
@@ -1142,4 +1145,17 @@ resident-transliteration-workaround-for-eng-fra=eng-hin,hin-fra
...
+mosip.resident.reg-processer-credential-partner-policy-url=${config.server.file.storage.uri}registration-processor-credential-partners.json
+
+hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+## Keymanager service
+#Type of keystore, Supported Types: PKCS11, PKCS12, Offline, JCE
+mosip.kernel.keymanager.hsm.keystore-type=OFFLINE
+# For PKCS11 provide Path of config file.
+# For PKCS12 keystore type provide the p12/pfx file path. P12 file will be created internally so provide only file path & file name.
+# For Offline & JCE property can be left blank, specified value will be ignored.
+mosip.kernel.keymanager.hsm.config-path=/config/softhsm-application.conf
```
Comment (vishwa-vyom) [inline resident-default.properties:1157]: if used offline, do you still need this property ? Also why we are using keymanager in resident service ?

File: resident-default.properties [hunk 8]
```diff
@@ -1142,4 +1145,17 @@ resident-transliteration-workaround-for-eng-fra=eng-hin,hin-fra
...
+# Passkey of keystore for PKCS11, PKCS12
+# For Offline & JCE proer can be left blank. JCE password use other JCE specific properties.
+mosip.kernel.keymanager.hsm.keystore-pass=${softhsm.kernel.security.pin}
+mosip.identity.get.excluded.attribute.list=UIN,verifiedAttributes,IDSchemaVersion
```
Comment (vishwa-vyom) [inline resident-default.properties:1161]: Is this property relevant to resident ? Looks more of idrepo property.

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: mosip-vid-policy.json (hunk 3 above)
Comment (kameshsr) [reply mosip-vid-policy.json:26]: This is to fix automation failure for generating vid for one time use vid. Actually, I have observed if the transaction allowed is 1 then it's allowing 0 times means no transaction is allowed and if transaction is 2 then only 1 transaction is allowed. so currently if transaction allowed is 1 then it will not allow to generate vid causing automation failure.

File: resident-default.properties (hunk 5 above)
Comment (kameshsr) [reply resident-default.properties:351]: As part of java21 migration changed this. for proxy api 500 internal server error was coming in case of invalid or empty path variable in api so to fix that done this change.

File: resident-default.properties (hunk 7 above)
Comment (kameshsr) [reply resident-default.properties:1157]: In resident service we are using keygenerator to generate keys 
https://github.com/mosip/resident-services/blob/772b2a4639fc745128828760c998e2f0f424f8e6/resident/resident-service/src/main/java/io/mosip/resident/service/impl/IdAuthServiceImpl.java#L314
So as part of java 21 migration it's not able to create bean  so added https://github.com/mosip/resident-services/blob/772b2a4639fc745128828760c998e2f0f424f8e6/resident/resident-service/src/main/java/io/mosip/resident/ResidentBootApplication.java#L47 multiple classes in import so KeyGenerator class needs these configs to be present. so added.

File: resident-default.properties (hunk 8 above)
Comment (kameshsr) [reply resident-default.properties:1161]: https://mosip.atlassian.net/browse/MOSIP-34475 Added this check as part of this bug.

File: ida-partner.cer (hunk 2 above)
Comment (kameshsr) [reply ida-partner.cer]: https://mosip.atlassian.net/browse/DSD-1769 Long back @loganathan-sekaran added this will check with @balaji-alluru for more info.

File: resident-default.properties (hunk 6 above)
Comment (kameshsr) [reply resident-default.properties]: Corrected template will add PR for mosip-data also.

File: partner-management-default.properties (hunk 4 above)
Comment (kameshsr) [reply partner-management-default.properties]: Requested @mayuradesh to review this.

File: resident-default.properties (hunk 6 above)
Comment (kameshsr) [reply resident-default.properties]: https://github.com/mosip/mosip-data/pull/315

File: partner-management-default.properties (hunk 4 above)
Comment (mayuradesh) [reply partner-management-default.properties]: This is only needed for revamp code

Comment (kameshsr) [reply partner-management-default.properties]: Removed.

File: ida-partner.cer (hunk 2 above)
Comment (kameshsr) [reply ida-partner.cer]: Removed file as not able to find why it's added in develop and dev-integration config.

File: mosip-vid-policy.json (hunk 3 above)
Comment (ghost) [reply mosip-vid-policy.json:26]: Then its a bug @kameshsr do we have a ticket for this issue?

Comment (kameshsr) [reply mosip-vid-policy.json:26]: https://mosip.atlassian.net/browse/MOSIP-35282
